src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -59,14 +69,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     String fileName = filePart.getSubmittedFileName();
     InputStream fileInputStream = filePart.getInputStream();
 
-    // Upload the file and get its URL
+    // Upload the image and get its URL
     String imageURL;
     boolean hasImage;
+    ArrayList<String> imageTags = new ArrayList<String>();
     
+    // Upload the image and save metadata if provided
     try{
+        byte[] imageBytes = fileInputStream.readAllBytes();
         imageURL = uploadToCloudStorage(fileName, fileInputStream);
         hasImage = true;
+
+        // Get the labels of the image that the user uploaded.
+        List<EntityAnnotation> imageLabels = getImageLabels(imageBytes);
```

Comment:
You can directly inline this into the for loop. so:

for(EntityAnnotation label : getImageLabels(imageBytes)) {
...
}